pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -212,6 +225,14 @@ func healthCheckTargets(targets []target) (int, []target, []time.Duration, []err
 
 		if nextCheck > 0 {
 			glog.V(3).Infof("Reconciling %s: is likely to go unhealthy in %v", t.string(), nextCheck)
+			recorder.Eventf(
+				&t.Machine,
+				corev1.EventTypeNormal,
+				healthcheckingv1alpha1.EventDetectedUnhealthy,
+				"Machine %v has unhealthy node %v",
+				t.Machine.Name,
```

Comment:
small nit: can we use `GetName()`?

we could also use `t.string()` here which will give us the mhc triggering this as well, have you consider it?

Reply:
I've considered it but I didn't like it at first because event is mainly about Machine object and not 'target'. But I like your point about MHC instance. So I changed it.